enterprise/cmd/frontend/internal/auth/githuboauth/provider.go

```diff
@@ -80,5 +87,20 @@ func requestedScopes(p *schema.GitHubAuthProvider) []string {
 		scopes = append(scopes, "read:org")
 	}
 
+	// Append extra scopes and ensure there are no duplicates
+	for _, s := range extraScopes {
+		var found bool
+		for _, inner := range scopes {
+			if inner == s {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			scopes = append(scopes, s)
+		}
+	}
```

Comment:
```suggestion
matchExtraScopes:
	for _, s := range extraScopes {
		for _, inner := range scopes {
			if inner == s {
				break matchExtraScopes
			}
		}

		scopes = append(scopes, s)
	}
```

I think this can be improved, matter of taste of course.

Reply:
This logic is slightly different actually, I'm trying to merge the two slices while ensuring there are no duplicates.